src/analysis/Analyses.ts

```diff
@@ -68,10 +68,10 @@ import * as StateHistory from 'src/libs/state-history';
 import * as Style from 'src/libs/style';
 import * as Utils from 'src/libs/utils';
 import { canWrite, isAzureWorkspace, isGoogleWorkspace, isGoogleWorkspaceInfo } from 'src/libs/workspace-utils';
-import { AppDetails } from 'src/pages/workspaces/hooks/useAppPolling';
-import { CloudEnvironmentDetails } from 'src/pages/workspaces/hooks/useCloudEnvironmentPolling';
-import { InitializedWorkspaceWrapper, StorageDetails } from 'src/pages/workspaces/hooks/useWorkspace';
-import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer';
+import { wrapWorkspace } from 'src/workspaces/container/WorkspaceContainer';
+import { AppDetails } from 'src/workspaces/hooks/useAppPolling';
+import { CloudEnvironmentDetails } from 'src/workspaces/hooks/useCloudEnvironmentPolling';
+import { InitializedWorkspaceWrapper, StorageDetails } from 'src/workspaces/hooks/useWorkspace';
```

Comment:
hmmm.... I think naming things based on their 'function' versus them happening to be a 'hook' might be more durable over time...  dusting off the directory tree proposal section .... looks like src/workspaces/shared/state would be where most of these hooks should go...

```
src/feature-area-1/
	bigFeatureExperience1/
		BigFeaturePanel/
      SubComponentA/…
	    SubComponentB/…
	    BigFeaturePanel.ts
	    BigFeaturePanel.test.ts
			useBigFeaturePanelState.ts
			useBigFeaturePanelState.test.ts
	shared/
		components/
    state/
		  BigFeatureStateFoo/
		    useBigFeatureStateFoo.ts
		    useBugFeatureStateFoo.test.ts
		providers/
			BigFeatureProviderThingy
			 	BigFeatureProviderThingy.ts
			 	BigFeatureProviderThingy.test.ts
src/shared/components/…
src/shared/data-providers/
	GoogleStorage/
		GoogleStorage.ts
		GoogleStorage.test.ts
```

Reply:
So are you thinking that anything that is used outside of `src/pages/workspaces` and `src/workspaces` goes into `src/workspaces/shared`? Which would mean that some hooks go into `src/workspaces/shared/hooks`, while the ones that are only in support of code in `src/workspaces` would be in either `src/workspace/state` (if used by multiple components within `src/workspaces` or `src/workspaces/container/state` (if used by only code within `container`, for example)?

Reply:
I'm unclear if "shared" refers to "shared outside of `src/workspaces`", or if it means "shared between components within `src/workspaces`". The tree you included has "state" directly under "src/feature-area-1".

Reply:
The eventual goal is to move to a package (or packages). Part of the appeal of packages is that the internal organization is hidden from external consumers (everything is imported from `@terra-ui-packages/<package>`), allowing teams to organize their code however they like without affecting others.

Reply:
Right, but I also don't know if we would want to put **all** of this code into a package. We may only want to put the parts that others depend on, in which case having it split up some within this directory structure would help us sort that out.

Reply:
Or perhaps we will have 2 packages-- one with shared code, and one with workspaces "private" code? I'm not sure if we care about reducing the size/scope of shared packages.

Reply:
Even for code that's not necessarily shared outside of Terra UI, I think we'll want to shift toward packages for the more well defined interfaces between feature areas.

Also, AoU re-implements a lot of Terra UI, so it may be worth sharing high level components to avoid duplication there.

>  I'm not sure if we care about reducing the size/scope of shared packages.
I'm not particularly concerned about it. Tree shaking in the consumer's build process allows a consumer to use part of a package without pulling in the whole thing. There's definitely a balance to be struck between smaller packages allowing precise updates vs larger packages making dependency management easier.

Reply:
If we are thinking that we will have large packages that hide the internal structure, then I would rather keep all hooks within a single "hooks" folder because I think it makes it easier for workspaces maintainers to understand and reuse code. Having to move things between a "shared" folder and other folders as their external consumption changes seems like an unnecessary burden, and if one has to dig into each directory structure to see if a hook has already been written makes it more likely that code will be duplicated.

What do you think, @msilva-broad?

Reply:
i think "shared" is maybe a loaded term... the intent was "shared within the feature area" or within the part of the tree that /shared shows up in.  It was not intended to signify any relation to cross-team code sharing.

And I do agree with the sentiment that packages will make team-by-team internal structure less of an issue.  I guess what I am hoping to discourage is another src/libs situation where things get dumped into one big heap...  

To pull in the article that Nick referenced, the primary folder structure should be around feature.... and if these hooks have broad usage within the feature area, then sure, put them near the top.... but naming it /state instead of /hooks would also let you put in the withWorkspace wrapper thing that for the time being is less convenient to use then hooks, but serve the same goal of providing state related to retrieved data.

src/workspaces/shared/state also helps avoid the misunderstanding that state should be separate from visual components... co-location is the default... by having /shared in there, it hopefully makes things more explicit.  src/workspaces/useMyHook.ts also works but gets cluttered quickly if there are a lot of things at that level.

Reply:
also, having some consistency between the feature areas will help when teams/devs need/want to hop around to help/contribute to other areas...

Reply:
chatted with Christina in Slack... 

Looks like things will setting into src/workspaces/common/state/... src/workspaces/common/WorkspaceAcl and so on.

Reply:
@msilva-broad and I agreed (over Slack) that "shared" is too overloaded and confusing. Therefore we are going with "common". I just pushed up individual commits to reorganize the files, and will squash them with their original commits if this looks good to folks (FYI @nawatts).

Here is what the directory structure looks like now. If a directory isn't expanded, it doesn't include a "state" directory.
![image](https://github.com/DataBiosphere/terra-ui/assets/484484/34fb1124-cfa7-4189-98a1-f81ba7d9e819)

Remainder (couldn't fit in one screen):
![image](https://github.com/DataBiosphere/terra-ui/assets/484484/b5e0fd24-0555-49ed-8942-010699b24583)

Reply:
Rebasing to group the cleanup commits with their original ones was too difficult wrt conflicts. :(

I have updated the PR description though to reflect this final design.